sdk/eventhub/Azure.Messaging.EventHubs.Processor/samples/Sample10_RunningInAzureStack.cs

```diff
@@ -0,0 +1,202 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.Messaging.EventHubs.Consumer;
+using Azure.Messaging.EventHubs.Processor.Samples.Infrastructure;
+using Azure.Messaging.EventHubs.Producer;
+using Azure.Storage.Blobs;
+
+namespace Azure.Messaging.EventHubs.Processor.Samples
+{
+    /// <summary>
+    ///   An example of running the Event Processor in the Azure Stack platform.
+    /// </summary>
+    ///
+    public class Sample10_RunningInAzureStack : IEventHubsBlobCheckpointSample
+    {
+        /// <summary>
+        ///   The name of the sample.
+        /// </summary>
+        ///
+        public string Name => nameof(Sample10_RunningInAzureStack);
+
+        /// <summary>
+        ///   A short description of the sample.
+        /// </summary>
+        ///
+        public string Description => "An example of running the Event Processor in the Azure Stack platform.";
+
+        /// <summary>
+        ///   Runs the sample using the specified Event Hubs and Azure storage connection information.
+        /// </summary>
+        ///
+        /// <param name="eventHubsConnectionString">The connection string for the Event Hubs namespace that the sample should target.</param>
+        /// <param name="eventHubName">The name of the Event Hub, sometimes known as its path, that the sample should run against.</param>
+        /// <param name="blobStorageConnectionString">The connection string for the storage account where checkpoints and state should be persisted.</param>
+        /// <param name="blobContainerName">The name of the blob storage container where checkpoints and state should be persisted.</param>
+        ///
+        public async Task RunAsync(string eventHubsConnectionString,
+                                   string eventHubName,
+                                   string blobStorageConnectionString,
+                                   string blobContainerName)
+        {
+            // The latest Azure Storage API supported with Azure Stack Hub version 2002 is not compatible with the one that the Azure
+            // Storage SDK officially supports.  For this reason, the Event Processor Client, which makes use of the Azure Storage
+            // Blobs SDK, cannot be used with Azure Stack resources by default.  This sample illustrates how to work around this issue
+            // by using a pipeline policy to force the Azure Storage SDK to downgrade its version.
```

Comment:
Avoid references mentioning that the version of the Storage service on Stack “is not compatible with the one that the Azure Storage SDK officially supports.” I know what you mean, but more accurately is to explaining the available version of the Storage service in different environments or platforms. There is just one Storage service and it happens that on Azure Stack there is another version available (not even mention a “previous version” because that might not be true in the future). That is different to painting the idea that the version of the Storage service on Stack…is not compatible with….Azure Storage SDK. That sounds like the Service storage is not compatible with itself.

Reply:
Done. I have changed the wording a bit.